util/generate_man.js

```diff
@@ -31,75 +36,75 @@ program
-	if (debug) {
-		console.debug('Processing file:', f)
+const processDovecotMd = () => {
+	return tree => {
+		return map(tree, (node) => {
+			if (node.value) {
+				node.value = node.value.replace(includesDM, (m, m1) => {
+					if (!m1.length) return m
+
+					const parts = m1.split(',').map((x) => x.trim())
+					switch (parts[0]) {
+					case 'man':
+						return parts[1] + '(' + (parts[3] ? parts[3] : '1') + ')'
+					case 'rfc':
+						return "RFC " + parts[1]
+					case 'setting':
+						return '`' + parts[1] + '`'
+					default:
+						return m1
+					}
+				})
+			}
+			return node
+		})
 	}
+}
 
-	/* Load base man file. */
-	const str = fs.readFileSync(f, 'utf8')
-	const content = matter(str).content
-
-	// https://pandoc.org/MANUAL.html#extension-pandoc_title_block
-	const fparts = path.basename(f).split('.')
-	let raw_md = "% " + fparts[0] + "(" + fparts[1] + ") " + gitHash + " | Dovecot\n" +
-		"%\n" +
-		"% " + dayjs().format('YYYY/MM/DD') + "\n\n"
-
-	/* Handle @include statements */
-	raw_md += processIncludes(content, f)
-
-	/* Process Dovecot markdown */
-	raw_md = raw_md.replace(includesDM, (m, m1) => {
-		if (!m1.length) return m
-
-		const parts = m1.split(',').map((x) => x.trim())
-		switch (parts[0]) {
-		case 'man':
-			return parts[1] + '(' + (parts[3] ? parts[3] : '1') + ')'
-
-		case 'setting':
-			return '`' + parts[1] + '`'
-
-		default:
-			return m1
-		}
-	})
+const main = async (path, outPath) => {
+	/* Create output directory, if it doesn't exist. */
+	if (!fs.existsSync(outPath)) {
+		await fs.promises.mkdir(outPath)
+	}
 
-	pdc(raw_md, 'markdown', 'man', [ '-s' ], (err, result) => {
-		if (err) throw err
-		const out_f = path.join(outPath, path.basename(f, '.md'))
-		fs.writeFileSync(out_f, result)
-		if (debug) {
-			console.debug('Man file written:', out_f)
-		}
-	})
-}
+	/* Generate list of man files. */
+	const files = (await manFiles()).flatMap((x) => fg.sync(x))
 
-/* Process @include statements (handles embedded includes) */
-function processIncludes(data, f) {
-	return data.replace(includesRE, (m, m1) => {
-		if (!m1.length) return m
+	/* Get hash of last git commit. */
+	const gitHash = gitCommitInfo().shortHash
 
-		const inc_f = path.join(path.dirname(f), m1)
+	/* Process man files. */
+	for (const f of files) {
 		if (debug) {
-			console.debug('    Include:', inc_f)
+			console.debug('Processing file:', f)
 		}
 
-		return processIncludes(matter(
-			fs.readFileSync(inc_f, 'utf8')
-		).content, inc_f)
-	})
+		/* Load base man file. */
+		const out_f = path.join(outPath, path.basename(f, '.md'));
+		const str = await fs.promises.readFile(f, 'utf8')
+		const content = doInclude(matter(str).content, f)
+
+		const fparts = path.basename(f).split('.')
+		const result = await unified().
+			use(remarkParse).
+			use(processDovecotMd).
+			use(remarkGfm).
+			use(remarkMan, {
+				manual: 'Dovecot',
+				section: parseInt(fparts[1]),
+				version: gitHash,
+				name: fparts[0].toUpperCase()
```

Comment:
The name header is duplicated in the output.

i.e., man ./doveadm-who.1:

```
DOVEADM-WHO(1)                      Dovecot                     DOVEADM-WHO(1)

NAME
       DOVEADM-WHO - doveadm-who

NAME
       doveadm-who - Show who is logged into the Dovecot server

SYNOPSIS
```

Reply:
that's because the file is wrong. i didn't yet fix all the manpages.

Reply:
oh right.

Reply:
It does seem to want you to not include a Name header in the markdown code, and instead use the argument parameter.

Seems like it requires the page to look like:

```
# doveadm-who(1) -- Show who is logged into the Dovecot server
```

So the documentation pages would no longer have "NAME" block.

Reply:
this seems to be correct behavior by man